File: Estruturais/Adapter/adapter01.py
# OBJETO
# intefcace client             
class ITarget:
    def request(self):
        pass

# cliente
class Client:
    def __init__(self, target):
        self.target = target

    def make_request(self):
        print("fazendo uma requisição")
        self.target.request()

#serviço existente
class Adaptee:
    def specific_request(self):
        print("request especifico do adaptee")


#Adaptador
class Adapter(ITarget):
    def __init__(self, adaptee):
        self.adaptee = adaptee

    def request(self):
        self.adaptee.specific_request()


#utilizando o adaptador
adaptee = Adaptee()
adapter = Adapter(adaptee)
client = Client(adapter)        

client.make_request()

File: Estruturais/Adapter/adapter01.js
// OBJETO
// Interface do cliente
class ITarget{
    request() {

    }
}

// Cliente
class Client{
    constructor(target){
        this.target = target;
    }

    makeRequest(){
        console.log("Fazendo uma requisição");
        this.target.request();
    }
}

// servico existente
class Adaptee{
    specficRequest(){
        console.log("Requisição específica do Adaptee.");
    }
}

// adaptador
class Adapter{
    constructor(adaptee){
        this.adaptee = adaptee;
    }

    request(){
        this.adaptee.specficRequest();
    }
}

// Utilização
// fazendo uma requisição
// requisição especifica do adaptee
const adaptee = new Adaptee();
const adapter = new Adapter(adaptee);
const client = new Client(adapter);

client.makeRequest(); 
